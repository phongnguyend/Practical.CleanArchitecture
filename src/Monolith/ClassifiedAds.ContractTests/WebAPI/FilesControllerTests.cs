using ApprovalTests;
using ApprovalTests.Reporters;
using ClassifiedAds.Application.AuditLogEntries.DTOs;
using ClassifiedAds.WebAPI.Models.Files;
using NJsonSchema;
using Xunit;

namespace ClassifiedAds.ContractTests.WebAPI
{
    [UseReporter(typeof(VisualStudioReporter))]
    public class FilesControllerTests
    {
        [Fact]
        public void FileEntryModelTest()
        {
            var schema = JsonSchema.FromType<FileEntryModel>().ToJson();
            Approvals.Verify(schema);
        }

        [Fact]
        public void UploadFileTest()
        {
            var schema = JsonSchema.FromType<UploadFileModel>().ToJson();
            Approvals.Verify(schema);
        }

        [Fact]
        public void AuditLogEntryDTOTest()
        {
            var schema = JsonSchema.FromType<AuditLogEntryDTO>().ToJson();
            Approvals.Verify(schema);
        }
    }
}
