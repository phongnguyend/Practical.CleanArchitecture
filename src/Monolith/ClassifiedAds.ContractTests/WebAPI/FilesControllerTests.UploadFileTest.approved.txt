{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "UploadFileModel",
  "type": "object",
  "additionalProperties": false,
  "required": [
    "FormFile"
  ],
  "properties": {
    "Name": {
      "title": "Name",
      "type": [
        "null",
        "string"
      ],
      "maxLength": 50,
      "minLength": 0
    },
    "Description": {
      "title": "Description",
      "type": [
        "null",
        "string"
      ],
      "maxLength": 50,
      "minLength": 0
    },
    "FormFile": {
      "title": "File",
      "type": "string",
      "format": "binary",
      "minLength": 1
    },
    "Encrypted": {
      "type": "boolean"
    }
  }
}